src/storage/LockingResourceStore.ts

```diff
@@ -55,4 +54,24 @@ export class LockingResourceStore implements AtomicResourceStore {
       await lock.release();
     }
   }
+
+  private async lockedRunGet(identifier: ResourceIdentifier, preferences: RepresentationPreferences,
+    conditions?: Conditions): Promise<Representation> {
+    const lock = await this.locks.acquire(identifier);
+
+    // Execute the getRepresentation function and return the resulting Representation.
+    let result;
+    try {
+      result = await this.source.getRepresentation(identifier, preferences, conditions);
+      return result;
+    } finally {
+      // Wait for the end or error event to be called to release the lock if the result contains a valid Readable.
+      if (!result || !result.data) {
+        await lock.release();
+      } else {
+        result.data.on('end', async(): Promise<void> => lock.release());
+        result.data.on('error', async(): Promise<void> => lock.release());
```

Comment:
So if my comment above is followed, we will be creating 3 callbacks with `async(): Promise<void> => lock.release()`. My suggestion is to:
- declare one callback that is reused (to save some memory)
- remove the return typing and `async` on there (it is not used)
- check if it is okay to call `lock.release` multiple times (!) and have the tests cover that case

Reply:
It's me with another linter comment again. :D Regarding your scond point: the linter requires for functions that return Promises to always be async, and also requires all functions to have return typings defined.

Reply:
We can't just `any` callbacks?

Reply:
That is still allowed by the linter but then you have the Inrupt style guide highly discouraging the use of `any` unless you have a really good reason.

Reply:
I do like here how you can immediately see here that the end/error function will not immediately do its call since it contains a promise.

Reply:
That's in fact not correct 😄 so misleading, if anything.

Reply:
Not sure what you mean with not correct, the `release` call returns a promise so it will not be immediately released when the end function resolves, only when the release promise resolves.

Reply:
The release will be immediately triggered. The _completion_ of the release (or at least the _notification_ of that completion) will indeed be delayed, but no one is waiting for this (hence my remark that this typing does not matter and is in fact misleading).

Reply:
@smessie feel free to use `any` here, it's too small of a detail to keep going on about :D